components/Tel.Egram.Components.Application/ApplicationController.cs

```diff
@@ -142,6 +175,31 @@ private void GoToWorkspacePage()
             Model.AuthenticationModel = null;
         }
 
+        private void UpdateConnectionState(int state)
+        {
+            if (Model.AuthenticationModel == null)
+            {
+                var model = _authenticationActivator.Activate(ref _authenticationController);
+                Model.AuthenticationModel = model;
+            }
+
+            var stateTexts = new string[]
+            {
+                "Connecting...",
+                "Connecting to proxy...",
+                "Ready.",
+                "Updating...",
+                "Waiting for network..."
+            };
+
+            Model.ConnectionState = stateTexts[state];
+
+            _workspaceController?.Dispose();
+
+            Model.StartupModel = null;
```

Comment:
Why Workspace and Startup models are getting nulled by connection state changes? This causes white screen instead of chat content. I think updating ConnectionState should be enough.